### ISSUE TITLE
Searchmonkey keeps crashing randomly.

### ISSUE THREAD

**Issue by CHJ85** [Luc-Ami/searchmonkey#1]
Not sure what this error means, but could this be resolved somehow? Searchmonkey keeps crashing sometimes.

```
Qt: Session management error: networkIdsList argument is NULL
The X11 connection broke (error 1). Did the X11 server die?
malloc(): invalid size (unsorted)
```

**Comment by Luc-Ami** [comment]
Hello. In order to help me; can I have some details ?
Your Linux distro, and, if you ve made a deep search inside files (for example : looking for a string inside files). Do your directories have MS Word files (prior 2007) and so on.
Regards

PS : I've seen Qt : error : My version of Searchmonkey is Gtk3 based, perhaps are you using an old unsupported version of Searchmonkey ?

**Comment by Luc-Ami** [comment]
The Qt version is here :   [https://sourceforge.net/projects/searchmonkey/files/kSearchmonkey%20Qt%20%28KDE%29/](url)

**Comment by CHJ85** [comment]
Version of Linux: Ubuntu 22.04.
Qt version: 5.15.3-1
Searchmonkey version: 0.8.3-1-build2 (the one from Ubuntu's repo)
And no microsoft files. Only plain text files.
Also. It doesn't happen during search, but while I'm opening these text files.
And it only happens every once in awhile, like I said. At random.

**Comment by Luc-Ami** [comment]
Thanks for your reply. I don't know who have compiled this version. I have no idea, in order to help you, because 0.8.3 is an old version.

**Comment by Luc-Ami** [comment]
And this **0.8.3-1-build2** version is available as a snap or a .DEB (from Ubuntu official repo ?).
And, please, is it a QT or a GTK version of Searchmonkey ? I am very surprised by the Qt crash ...
regards

**Comment by Luc-Ami** [comment]
Another question : if I understand, you're searching a certain file, and, THEN, attempt to open it ; in this case, Searchmonkey requires (in option menu) the complete path to your predered text editor (/usr/bin/ ...).
Or, are you launched a "deep search" for a certain string INSIDE a text file ? In that case, the rifht panel must show an abstract of the text with the highlighted string).

**Comment by CHJ85** [comment]
No, I'm searching a directory of many files. Some times, when I open a file from the results list, Searchmonkey crashes.
It happens usually if there are many results in the list, like 50+ or something.
But I'll add /usr/bin/ to the launch option and see if that helps.

**Comment by CHJ85** [comment]
And according to Ubuntu, it's the latest version. And btw, it's the GTK+ version.
https://packages.ubuntu.com/search?keywords=searchmonkey&searchon=names&suite=all&section=all

**Comment by Luc-Ami** [comment]
Thanks for these various clarifications. BY the way, 0.8.3 is an old version of 2018.

**Comment by Luc-Ami** [comment]
I've just compiled this old 0.8.3 version : is it the same for you ? (please click link for my screenshot) 
![sm-Capture du 2022-10-27 20-05-35](https://user-images.githubusercontent.com/36115088/198366834-6b17548d-9d94-430d-92f1-d9171fc77724.png)

**Comment by CHJ85** [comment]
Yes, that's the one. Well. I did try adding the PPA (ppa:amiluc-bis/searchmonkey), but I was unable to due to inaccurate Ubuntu version.  However. I did manage to install the PPA's DEB file manually.
Do you think you could upstream the latest version to Ubuntu's repo?

**Comment by Luc-Ami** [comment]
I'm still using Ubuntu 16.04, so I can't update my PPA. Feel free to upload to Ubuntun repos or do your own PPA, Searchmonkey if GPL licensed.

**Comment by Luc-Ami** [comment]
A question : I've just noticed that the default editor (for Searchmonkey  0.8.3) can't edit files when path contains spaces, like this  
/usr/home/bla bla/myfile.txt

Perhaps, it's the same issue for you ?

**Comment by Luc-Ami** [comment]
I confirm that Searchmonkey 0.8.3 is bugged when path are containing spaces, not Searchmonkey 0.9, here is a screenshot for 0.9, my default editor (Gedit) opens the selected file.
![Capture du 2022-10-28 14-00-21](https://user-images.githubusercontent.com/36115088/198582577-16aac03f-a9a1-4525-a01c-e1bb73776f95.png)

**Comment by CHJ85** [comment]
Thanks. However, I'm actually using Kate as the default editor. So I don't know if that's the issue.
